modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Comments/MongoCommentRepository.cs

```diff
@@ -23,23 +23,28 @@ public MongoCommentRepository(IMongoDbContextProvider<ICmsKitMongoDbContext> dbC
 
         public async Task<CommentWithAuthorQueryResultItem> GetWithAuthorAsync(Guid id, CancellationToken cancellationToken = default)
         {
-            var query = from comment in (await GetMongoQueryableAsync(cancellationToken))
-                join user in (await GetDbContextAsync(cancellationToken)).CmsUsers on comment.CreatorId equals user.Id
-                where id == comment.Id
-                select new CommentWithAuthorQueryResultItem
-                {
-                    Comment = comment,
-                    Author = user
-                };
+            var dbContext = await GetDbContextAsync();
+            var commentQueryable = await GetMongoQueryableAsync(cancellationToken);
+            var userQueryable = dbContext.Collection<CmsUser>();
+
+            var joined = commentQueryable.Join(userQueryable, x => x.CreatorId, x => x.Id, (comment, user) => new
```

Comment:
We can also use the query version instead of the query version.  I think it is more readable.